lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -468,6 +468,18 @@ public void checkCanRenameMaterializedView(ConnectorSecurityContext context, Sch
         }
     }
 
+    @Override
+    public void checkCanGrantExecuteFunctionPrivilege(ConnectorSecurityContext context, FunctionKind functionKind, SchemaRoutineName functionName, TrinoPrincipal grantee, boolean grantOption)
+    {
+        switch (functionKind) {
+            case SCALAR, AGGREGATE, WINDOW:
+                return;
+            case TABLE:
+                denyExecuteFunction(functionName.toString());
+        }
+        throw new UnsupportedOperationException("Unsupported function kind: " + functionKind);
```

Comment:
I am under impression that this throw is not needed. However, it leaves no question that is a dead code.

Or maybe you can use `return switch` here?

Reply:
I wanted to say the same, but the return value is `void` so it won't work.